### PR TITLE
[skip ci] fix broken ceph-fetch-keys role

### LINKED ISSUE #6072: take-over-existing-cluster.yml fails if fetch_directory not set
**Bug Report**

What happened: take-over-existing-cluster.yml fails because fetch_directory was not set

What you expected to happen: 
* fetch directory creates as per default and then used

How to reproduce it (minimal and precise):
* simply run take-over-existing-cluster.yml on any ceph-cluster without setting the fetch_directory variable

Problem: 
* in roles/ceph-fetch-keys/tasks/main.yml a fetch directory is first created via fetch_directory variable or by a default('fetch/') 
* afterwards in the same role only the variable is used

simple fix: use {{ fetch_directory | default('fetch/') }} instead of {{ fetch_directory }} everywhere or set the variable to it default if it isn't set already

I don't think there is any need for inventory, group_vars or log files here

**Environment**:
* OS ubuntu: 16.04
* Kernel:  4.4.0-186-generic #216-Ubuntu SMP Wed Jul 1 05:34:05 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
* Ansible version: ansible-playbook 2.9.15
* ceph-ansible version:  git head
* Ceph version: ceph version 10.2.11 (e4b061b47f07f583c92a050d9e84b1813a35671e) (I know embarrassing ...)

### PR DESCRIPTION
**Pull request by scriptkiddy**
set fetch_directory variable in default/main.yml instead of using the defaults jinja filter in tasks/main.yml

Fixes: #6072 

Signed-off-by: Karl-Heinz Preuß karl-heinz.preuss@cms.hu-berlin.de

### CODE REVIEW

**Comment by guits** [comment]
hey @scriptkiddy 

you are missing a commit message and a sign off here

File: roles/ceph-fetch-keys/tasks/main.yml [hunk 1]
```diff
@@ -11,10 +11,10 @@
   delegate_to: localhost
   become: false
 
-- name: "copy ceph user and bootstrap keys to the ansible server in {{ fetch_directory }}/{{ fsid }}/"
+- name: "copy ceph user and bootstrap keys to the ansible server in {{ fetch_directory | default('fetch/') }}/{{ fsid }}/"
   fetch:
     src: "{{ item }}"
-    dest: "{{ fetch_directory }}/{{ fsid }}/{{ item }}"
+    dest: "{{ fetch_directory | default('fetch/') }}/{{ fsid }}/{{ item }}"
```
Comment (guits) [inline roles/ceph-fetch-keys/tasks/main.yml:17]: it's probably better to add back a `fetch_directory` variable in `./roles/ceph-fetch-keys/defaults/main.yml`

Comment (scriptkiddy) [reply roles/ceph-fetch-keys/tasks/main.yml:17]: Ok I'll do that. Wasn't sure what solution you preferred. Will take another day though.

**Comment by guits** [comment]
@scriptkiddy you have to run the `./generate_group_vars_sample.sh` script
also, could you squash your commits please?

thanks!

**Comment by scriptkiddy** [comment]
Hi @guits 

> @scriptkiddy you have to run the `./generate_group_vars_sample.sh` script
Okay I did that. I then used the sample file as ceph-fetch-keys/defaults/main.yml and added the sample file to the git repo I hope that is what you wanted.

> also, could you squash your commits please?

apparently I'm a bit of a git illiterate. I hope I got it right now.

cheers

> 
> thanks!

**Review by guits — DISMISSED** [review]

**Comment by dsavineau** [comment]
jenkins test syntax check

**Comment by dsavineau** [comment]
jenkins test playbook syntax

**Review by guits — APPROVED** [review]

**Pull request merged**